Components/UploadData/index.js

```diff
@@ -0,0 +1,86 @@
+import Heading from "../../Layout/Heading";
+import Paragraph from "../../Layout/Paragraph";
+import CenteredDiv from "../../Layout/CenteredDiv";
+import CorrectArrays from "../CorrectArrays";
+import FileUploader from "../FileUploader";
+import { useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+const { convertCSVToArray } = require("convert-csv-to-array");
+import { StyledInputTypeSubmit } from "@/Layout/StyledInputTypeSubmit";
+
+export default function UploadData() {
+  //State to store keys from the CSV file
+  const [keynames, setKeynames] = useLocalStorageState("keynames", {
+    defaultValue: [],
+  });
+
+  //State to store the values
+  const [vals, setVals] = useLocalStorageState("vals", {
+    defaultValue: [],
+  });
+
+  //State to store the file object
+  const [fileObj, setFileObj] = useState({});
+
+  // State to store the isUploaded status
+  const [isUploaded, setIsUploaded] = useState(false);
+
+  const handleFile = (file) => {
+    setFileObj(file);
+    setIsUploaded(true);
+  };
+
+  // Name of file
+  const fileName = fileObj.name;
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+  };
+
+  // Reading the file and converting the result to an array of objects
+  const handleConversion = () => {
+    const reader = new FileReader();
+
+    reader.onload = function (e) {
+      const arrayOfObjects = convertCSVToArray(e.target.result, {
+        separator: ",",
+      });
+
+      // The output of the convertCSVToArray function needs further processing
+      const [keys, ...correctValues] = CorrectArrays(arrayOfObjects);
+
+      setKeynames(keys);
+      setVals(correctValues);
+      alert("File processed successfully");
+    };
+
+    reader.readAsText(fileObj);
+  };
```

Comment:
I think you could easily combine these two functions :) 
Also better to add a little error handling :)

```suggestion
function handleSubmitAndConvert (event) {
  event.preventDefault(); 

  if (!fileObj) {
    alert("No file selected!");
    return;
  }

  const reader = new FileReader();

  reader.onload = function (loadEvent) {
    const arrayOfObjects = convertCSVToArray(loadEvent.target.result, {
      separator: ",",
    });

    const [keys, ...correctValues] = CorrectArrays(arrayOfObjects);

    setKeynames(keys);
    setVals(correctValues);
    alert("File processed successfully");
  };

  reader.readAsText(fileObj);
};
```

Reply:
The alert "No file selected" is not needed because it is not possible to submit and convert the file before the file was uploaded. Because the Submit button is disabled until the upload is done. So we have not implemented this change.

With the single function solution there is an error - "Unhandled Runtime Error
TypeError: Failed to execute 'readAsText' on 'FileReader': parameter 1 is not of type 'Blob'." So we have not implemented this change.